pydrive2/files.py

```diff
@@ -518,6 +518,36 @@ def Delete(self, param=None):
         """
         self._FilesDelete(param=param)
 
+    @LoadAuth
+    def Copy(self, target_folder, new_title = None, param=None):
```

Comment:
style: `new_title=None`

Reply:
Question: how do I copy to the root of my personal drive or to the root of a shared drive?

Reply:
you can't copy a folder, if a user attempt that they will receive:

```python
[WARNING] googleapiclient.http -> _should_retry_response: Encountered 403 Forbidden with reason "cannotCopyFile"
Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 547, in Copy
    self.auth.service.files().copy(**param).execute()
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/http.py", line 938, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 403 when requesting https://www.googleapis.com/drive/v2/files/1KL7v_9MSReLZifHhpv8mJG_NY1q9JAXC/copy?supportsAllDrives=true&alt=json returned "This file cannot be copied by the authenticated user.". Details: "[{'domain': 'global', 'reason': 'cannotCopyFile', 'message': 'This file cannot be copied by the authenticated user.'}]">

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/tests/main.py", line 66, in <module>
    file.Copy(folder)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/auth.py", line 82, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 549, in Copy
    raise ApiRequestError(error)
pydrive2.files.ApiRequestError: <HttpError 403 when requesting https://www.googleapis.com/drive/v2/files/1KL7v_9MSReLZifHhpv8mJG_NY1q9JAXC/copy?supportsAllDrives=true&alt=json returned "This file cannot be copied by the authenticated user.". Details: "[{'domain': 'global', 'reason': 'cannotCopyFile', 'message': 'This file cannot be copied by the authenticated user.'}]">
```

To copy a folder you need a recursive function that copy each file and create the folders as needed.

Should we check if the file they are attempting to copy is a folder and raise an error?

Reply:
same way as any other folder

```python
folder = drive.CreateFile({'id': 'root'})

file.Copy(folder)  
```

i don't know about shared drives btw, i can't do tests on that so i don't know if things are different.

Reply:
> What I was asking about is - what should I pass to this method to copy a _file_ to the _root_ of the Google Drive? or to a shared Google Drive?

sorry, yesterday i read that the wrong way and the answer was off-topic, so i deleted the comment and did a new one with the real answer (see comment above)